sphinx_immaterial/mermaid_diagrams.py

```diff
@@ -54,13 +61,38 @@ def depart_mermaid_node_latex(self, node: mermaid_node):
     self.body.append("\n\\end{sphinxVerbatim}\n")
 
 
+def on_builder_init(app: Sphinx):
+    """Init the copy-mermaid-dist flag to False"""
+    setattr(app.env, _COPY_MERMAID_DIST_ENV_KEY, False)
+
+
+def copy_mermaid_dist(app: Sphinx, env: BuildEnvironment):
+    if app.builder.name not in ("html", "dirhtml"):
+        return  # mermaid src is only used in HTML output
+    if getattr(app.env, _COPY_MERMAID_DIST_ENV_KEY, False) is True:
+        # copy the mermaid dist file (if not already done)
+        dst = Path(app.outdir, "_static", "mermaid")
+        if dst.exists():
+            return
+        shutil.copytree(str(Path(__file__).parent / "bundles" / "mermaid"), dst)
+
+
+def _merge_env_key(
+    app: Sphinx, env: BuildEnvironment, docnames: List[str], other: BuildEnvironment
+) -> None:
+    setattr(env, _COPY_MERMAID_DIST_ENV_KEY, getattr(other, _COPY_MERMAID_DIST_ENV_KEY))
```

Comment:
This should do a "logical or" rather than just unconditionally copying the value.

Reply:
done